src/Microsoft.AspNetCore.Routing/Matching/CandidateSet.cs

```diff
@@ -177,6 +205,56 @@ internal CandidateSet(Candidate[] candidates)
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value which indicates where the <see cref="Http.Endpoint"/> is considered
+        /// a valid candiate for the current request. Set this value to <c>false</c> to exclude an
+        /// <see cref="Http.Endpoint"/> from consideration.
+        /// </summary>
+        public bool IsValidCandidate(int index)
+        {
+            // Friendliness for inlining
+            if ((uint)index >= Count)
+            {
+                ThrowIndexArgumentOutOfRangeException();
+            }
+
+            if (Count < BitVectorSize)
+            {
+                // Get the n-th bit
+                return _validity[0x00000001 << index];
+            }
+            else
+            {
+                return _largeCapactityValidity[index];
+            }
+        }
+
+        /// <summary>
+        /// Sets the validitity of the candidate at the provided index.
+        /// </summary>
+        /// <param name="index">The candidate index.</param>
+        /// <param name="value">
+        /// The value to set. If <c>true</c> the candidate is considered valid for the current request.
+        /// </param>
+        public void SetValidity(int index, bool value)
```

Comment:
No, I think that would add negative value, it would have to do a linear search. This API as it is right now might be a little clunky to use but it's very hard to misuse.